src/packet.rs

```diff
@@ -274,6 +275,16 @@ pub const BLOB_FLAG_IS_CODING: u32 = 0x1;
 pub const BLOB_HEADER_SIZE: usize = align!(BLOB_SIZE_END, 64);
 
 impl Blob {
+    pub fn new(data: &[u8]) -> Self {
+        let mut blob = Self::default();
+        let data_len = cmp::min(data.len(), blob.data.len());
```

Comment:
this won't always be zero?

Reply:
blob.data is a fixed size array of size 65408, so it shouldn't be

Reply:
ok, I see.  we're gonna assert!() that data is small enough to fit?

Reply:
Added a check where new() is called in erasure to make sure the data fits.